Use after free on CI

Complete log: https://pipelines.actions.githubusercontent.com/t0EA9Ce0xHPdewtBWuxvsFssNmk7cuE6RUJ1retNcOjkYhlBCR/_apis/pipelines/1/runs/307/signedlogcontent/10?urlExpires=2021-04-13T11%3A56%3A13.4141370Z&urlSigningMethod=HMACV1&urlSignature=oAb9OZGu3HAnb9RDj7QH4QnZeOjPA%2F2E6xsuNiuIxnI%3D.

Relevant part:
<details>
<code><pre>
2021-04-13T11:54:23.5928958Z failures:
2021-04-13T11:54:23.5937792Z 
2021-04-13T11:54:23.5939529Z ---- src/actor_ref/rpc.rs - actor_ref::rpc (line 86) stdout ----
2021-04-13T11:54:23.5940471Z Test executable failed (exit code 1).
2021-04-13T11:54:23.5941048Z 
2021-04-13T11:54:23.5941584Z stdout:
2021-04-13T11:54:23.5942202Z Increased count to 10
2021-04-13T11:54:23.5942696Z 
2021-04-13T11:54:23.5943245Z stderr:
2021-04-13T11:54:23.5943819Z =================================================================
2021-04-13T11:54:23.5945264Z ==5349==ERROR: AddressSanitizer: heap-use-after-free on address 0x604000002070 at pc 0x55fa25d9a601 bp 0x7f6ada7fb540 sp 0x7f6ada7fb538
2021-04-13T11:54:23.5946522Z WRITE of size 1 at 0x604000002070 thread T2 (Sync actor 1000)
2021-04-13T11:54:23.5947446Z     #0 0x55fa25d9a600  (/tmp/rustdoctestne7jWm/rust_out+0x342600)
2021-04-13T11:54:23.5948477Z     #1 0x55fa25b871ae  (/tmp/rustdoctestne7jWm/rust_out+0x12f1ae)
2021-04-13T11:54:23.5949492Z     #2 0x55fa25ba7846  (/tmp/rustdoctestne7jWm/rust_out+0x14f846)
2021-04-13T11:54:23.5950523Z     #3 0x55fa25b81e95  (/tmp/rustdoctestne7jWm/rust_out+0x129e95)
2021-04-13T11:54:23.5951533Z     #4 0x55fa25b6f53a  (/tmp/rustdoctestne7jWm/rust_out+0x11753a)
2021-04-13T11:54:23.5952548Z     #5 0x55fa25b9a5c8  (/tmp/rustdoctestne7jWm/rust_out+0x1425c8)
2021-04-13T11:54:23.5953558Z     #6 0x55fa25baebb8  (/tmp/rustdoctestne7jWm/rust_out+0x156bb8)
2021-04-13T11:54:23.5954601Z     #7 0x55fa25badbde  (/tmp/rustdoctestne7jWm/rust_out+0x155bde)
2021-04-13T11:54:23.5955664Z     #8 0x55fa25b90cd4  (/tmp/rustdoctestne7jWm/rust_out+0x138cd4)
2021-04-13T11:54:23.5956679Z     #9 0x55fa25b8fa8b  (/tmp/rustdoctestne7jWm/rust_out+0x137a8b)
2021-04-13T11:54:23.5957715Z     #10 0x55fa25b76d5e  (/tmp/rustdoctestne7jWm/rust_out+0x11ed5e)
2021-04-13T11:54:23.5959811Z     #11 0x55fa25b79926  (/tmp/rustdoctestne7jWm/rust_out+0x121926)
2021-04-13T11:54:23.5960907Z     #12 0x55fa25b6a8e9  (/tmp/rustdoctestne7jWm/rust_out+0x1128e9)
2021-04-13T11:54:23.5962306Z     #13 0x55fa25b7a1f2  (/tmp/rustdoctestne7jWm/rust_out+0x1221f2)
2021-04-13T11:54:23.5963273Z     #14 0x55fa25bb2bdc  (/tmp/rustdoctestne7jWm/rust_out+0x15abdc)
2021-04-13T11:54:23.5964361Z     #15 0x55fa25b79e8a  (/tmp/rustdoctestne7jWm/rust_out+0x121e8a)
2021-04-13T11:54:23.5965330Z     #16 0x55fa25b773a9  (/tmp/rustdoctestne7jWm/rust_out+0x11f3a9)
2021-04-13T11:54:23.5966260Z     #17 0x55fa25b792c8  (/tmp/rustdoctestne7jWm/rust_out+0x1212c8)
2021-04-13T11:54:23.5967212Z     #18 0x55fa25b7def5  (/tmp/rustdoctestne7jWm/rust_out+0x125ef5)
2021-04-13T11:54:23.5968146Z     #19 0x55fa25f916a9  (/tmp/rustdoctestne7jWm/rust_out+0x5396a9)
2021-04-13T11:54:23.5969074Z     #20 0x55fa25f91223  (/tmp/rustdoctestne7jWm/rust_out+0x539223)
2021-04-13T11:54:23.5969997Z     #21 0x55fa260468c1  (/tmp/rustdoctestne7jWm/rust_out+0x5ee8c1)
2021-04-13T11:54:23.5971301Z     #22 0x7f6ade127608  (/lib/x86_64-linux-gnu/libpthread.so.0+0x9608)
2021-04-13T11:54:23.5972451Z     #23 0x7f6addef9292  (/lib/x86_64-linux-gnu/libc.so.6+0x122292)
2021-04-13T11:54:23.5973052Z 
2021-04-13T11:54:23.5973958Z 0x604000002070 is located 32 bytes inside of 40-byte region [0x604000002050,0x604000002078)
2021-04-13T11:54:23.5974743Z freed by thread T1 (Worker 1) here:
2021-04-13T11:54:23.5975568Z     #0 0x55fa25b4a8f2  (/tmp/rustdoctestne7jWm/rust_out+0xf28f2)
2021-04-13T11:54:23.5976501Z     #1 0x55fa25e718dc  (/tmp/rustdoctestne7jWm/rust_out+0x4198dc)
2021-04-13T11:54:23.5977603Z [0m[0m[1m[31merror[0m[1m:[0m test failed, to rerun pass '--doc'
2021-04-13T11:54:23.5978495Z     #2 0x55fa25eba7c9  (/tmp/rustdoctestne7jWm/rust_out+0x4627c9)
2021-04-13T11:54:23.5982296Z make: *** [Makefile:28: test_sanitiser] Error 101
2021-04-13T11:54:23.5982841Z     #3 0x55fa25b9e261  (/tmp/rustdoctestne7jWm/rust_out+0x146261)
2021-04-13T11:54:23.5984456Z     #4 0x55fa25ba388e  (/tmp/rustdoctestne7jWm/rust_out+0x14b88e)
2021-04-13T11:54:23.5985647Z     #5 0x55fa25b9fcb5  (/tmp/rustdoctestne7jWm/rust_out+0x147cb5)
2021-04-13T11:54:23.5987860Z     #6 0x55fa25b8466b  (/tmp/rustdoctestne7jWm/rust_out+0x12c66b)
2021-04-13T11:54:23.5988623Z     #7 0x55fa25b7b876  (/tmp/rustdoctestne7jWm/rust_out+0x123876)
2021-04-13T11:54:23.5989222Z     #8 0x55fa25ba96b8  (/tmp/rustdoctestne7jWm/rust_out+0x1516b8)
2021-04-13T11:54:23.5989827Z     #9 0x55fa25baf9c1  (/tmp/rustdoctestne7jWm/rust_out+0x1579c1)
2021-04-13T11:54:23.5990424Z     #10 0x55fa25bad1c7  (/tmp/rustdoctestne7jWm/rust_out+0x1551c7)
2021-04-13T11:54:23.5991190Z     #11 0x55fa25b7a505  (/tmp/rustdoctestne7jWm/rust_out+0x122505)
2021-04-13T11:54:23.5991868Z     #12 0x55fa25cabc3a  (/tmp/rustdoctestne7jWm/rust_out+0x253c3a)
2021-04-13T11:54:23.5992564Z     #13 0x55fa25bb7cd4  (/tmp/rustdoctestne7jWm/rust_out+0x15fcd4)
2021-04-13T11:54:23.5993186Z     #14 0x55fa25bb6bec  (/tmp/rustdoctestne7jWm/rust_out+0x15ebec)
2021-04-13T11:54:23.5993974Z     #15 0x55fa25d044e8  (/tmp/rustdoctestne7jWm/rust_out+0x2ac4e8)
2021-04-13T11:54:23.5994603Z     #16 0x55fa25d18bef  (/tmp/rustdoctestne7jWm/rust_out+0x2c0bef)
2021-04-13T11:54:23.5995239Z     #17 0x55fa25d4084a  (/tmp/rustdoctestne7jWm/rust_out+0x2e884a)
2021-04-13T11:54:23.5995857Z     #18 0x55fa25cdeb1b  (/tmp/rustdoctestne7jWm/rust_out+0x286b1b)
2021-04-13T11:54:23.5996489Z     #19 0x55fa25d2824b  (/tmp/rustdoctestne7jWm/rust_out+0x2d024b)
2021-04-13T11:54:23.5997098Z     #20 0x55fa25d35049  (/tmp/rustdoctestne7jWm/rust_out+0x2dd049)
2021-04-13T11:54:23.5997727Z     #21 0x55fa25d3550c  (/tmp/rustdoctestne7jWm/rust_out+0x2dd50c)
2021-04-13T11:54:23.5998340Z     #22 0x55fa25d2842b  (/tmp/rustdoctestne7jWm/rust_out+0x2d042b)
2021-04-13T11:54:23.5998966Z     #23 0x55fa25cde52e  (/tmp/rustdoctestne7jWm/rust_out+0x28652e)
2021-04-13T11:54:23.5999584Z     #24 0x55fa25c5ac1e  (/tmp/rustdoctestne7jWm/rust_out+0x202c1e)
2021-04-13T11:54:23.6000211Z     #25 0x55fa25f91223  (/tmp/rustdoctestne7jWm/rust_out+0x539223)
2021-04-13T11:54:23.6000813Z     #26 0x55fa260468c1  (/tmp/rustdoctestne7jWm/rust_out+0x5ee8c1)
2021-04-13T11:54:23.6001818Z     #27 0x7f6ade127608  (/lib/x86_64-linux-gnu/libpthread.so.0+0x9608)
2021-04-13T11:54:23.6002363Z 
2021-04-13T11:54:23.6002757Z previously allocated by thread T1 (Worker 1) here:
2021-04-13T11:54:23.6003416Z     #0 0x55fa25b4aa3d  (/tmp/rustdoctestne7jWm/rust_out+0xf2a3d)
2021-04-13T11:54:23.6004041Z     #1 0x55fa25e71443  (/tmp/rustdoctestne7jWm/rust_out+0x419443)
2021-04-13T11:54:23.6004649Z     #2 0x55fa25eba5bf  (/tmp/rustdoctestne7jWm/rust_out+0x4625bf)
2021-04-13T11:54:23.6005288Z     #3 0x55fa25b9db1b  (/tmp/rustdoctestne7jWm/rust_out+0x145b1b)
2021-04-13T11:54:23.6005907Z     #4 0x55fa25b9de9a  (/tmp/rustdoctestne7jWm/rust_out+0x145e9a)
2021-04-13T11:54:23.6006537Z     #5 0x55fa25ba38ea  (/tmp/rustdoctestne7jWm/rust_out+0x14b8ea)
2021-04-13T11:54:23.6007150Z     #6 0x55fa25b6f035  (/tmp/rustdoctestne7jWm/rust_out+0x117035)
2021-04-13T11:54:23.6007775Z     #7 0x55fa25b99f1e  (/tmp/rustdoctestne7jWm/rust_out+0x141f1e)
2021-04-13T11:54:23.6008381Z     #8 0x55fa25b983cc  (/tmp/rustdoctestne7jWm/rust_out+0x1403cc)
2021-04-13T11:54:23.6009008Z     #9 0x55fa25bad1c7  (/tmp/rustdoctestne7jWm/rust_out+0x1551c7)
2021-04-13T11:54:23.6009628Z     #10 0x55fa25b7a505  (/tmp/rustdoctestne7jWm/rust_out+0x122505)
2021-04-13T11:54:23.6010258Z     #11 0x55fa25cabc3a  (/tmp/rustdoctestne7jWm/rust_out+0x253c3a)
2021-04-13T11:54:23.6010883Z     #12 0x55fa25bb7cd4  (/tmp/rustdoctestne7jWm/rust_out+0x15fcd4)
2021-04-13T11:54:23.6011532Z     #13 0x55fa25bb6bec  (/tmp/rustdoctestne7jWm/rust_out+0x15ebec)
2021-04-13T11:54:23.6012163Z     #14 0x55fa25d044e8  (/tmp/rustdoctestne7jWm/rust_out+0x2ac4e8)
2021-04-13T11:54:23.6012800Z     #15 0x55fa25d18bef  (/tmp/rustdoctestne7jWm/rust_out+0x2c0bef)
2021-04-13T11:54:23.6013417Z     #16 0x55fa25d4084a  (/tmp/rustdoctestne7jWm/rust_out+0x2e884a)
2021-04-13T11:54:23.6014045Z     #17 0x55fa25cdeb1b  (/tmp/rustdoctestne7jWm/rust_out+0x286b1b)
2021-04-13T11:54:23.6014657Z     #18 0x55fa25d2824b  (/tmp/rustdoctestne7jWm/rust_out+0x2d024b)
2021-04-13T11:54:23.6015274Z     #19 0x55fa25d35049  (/tmp/rustdoctestne7jWm/rust_out+0x2dd049)
2021-04-13T11:54:23.6015883Z     #20 0x55fa25d3550c  (/tmp/rustdoctestne7jWm/rust_out+0x2dd50c)
2021-04-13T11:54:23.6016615Z     #21 0x55fa25d2842b  (/tmp/rustdoctestne7jWm/rust_out+0x2d042b)
2021-04-13T11:54:23.6017346Z     #22 0x55fa25cde52e  (/tmp/rustdoctestne7jWm/rust_out+0x28652e)
2021-04-13T11:54:23.6017972Z     #23 0x55fa25c5ac1e  (/tmp/rustdoctestne7jWm/rust_out+0x202c1e)
2021-04-13T11:54:23.6018560Z     #24 0x55fa25f91223  (/tmp/rustdoctestne7jWm/rust_out+0x539223)
2021-04-13T11:54:23.6019229Z     #25 0x55fa260468c1  (/tmp/rustdoctestne7jWm/rust_out+0x5ee8c1)
2021-04-13T11:54:23.6020057Z     #26 0x7f6ade127608  (/lib/x86_64-linux-gnu/libpthread.so.0+0x9608)
2021-04-13T11:54:23.6020435Z 
2021-04-13T11:54:23.6020762Z Thread T2 (Sync actor 1000) created by T0 here:
2021-04-13T11:54:23.6021293Z     #0 0x55fa25b3500c  (/tmp/rustdoctestne7jWm/rust_out+0xdd00c)
2021-04-13T11:54:23.6021885Z     #1 0x55fa26046249  (/tmp/rustdoctestne7jWm/rust_out+0x5ee249)
2021-04-13T11:54:23.6022480Z     #2 0x55fa25b78064  (/tmp/rustdoctestne7jWm/rust_out+0x120064)
2021-04-13T11:54:23.6023073Z     #3 0x55fa25b79b16  (/tmp/rustdoctestne7jWm/rust_out+0x121b16)
2021-04-13T11:54:23.6023683Z     #4 0x55fa25b8efc3  (/tmp/rustdoctestne7jWm/rust_out+0x136fc3)
2021-04-13T11:54:23.6024281Z     #5 0x55fa25b8df60  (/tmp/rustdoctestne7jWm/rust_out+0x135f60)
2021-04-13T11:54:23.6024877Z     #6 0x55fa25b8e497  (/tmp/rustdoctestne7jWm/rust_out+0x136497)
2021-04-13T11:54:23.6025466Z     #7 0x55fa25b94ac2  (/tmp/rustdoctestne7jWm/rust_out+0x13cac2)
2021-04-13T11:54:23.6026076Z     #8 0x55fa25bb1c34  (/tmp/rustdoctestne7jWm/rust_out+0x159c34)
2021-04-13T11:54:23.6026669Z     #9 0x55fa25b7e41d  (/tmp/rustdoctestne7jWm/rust_out+0x12641d)
2021-04-13T11:54:23.6027280Z     #10 0x55fa25b771e3  (/tmp/rustdoctestne7jWm/rust_out+0x11f1e3)
2021-04-13T11:54:23.6027883Z     #11 0x55fa25ff9c7d  (/tmp/rustdoctestne7jWm/rust_out+0x5a1c7d)
2021-04-13T11:54:23.6028493Z     #12 0x55fa25e36aae  (/tmp/rustdoctestne7jWm/rust_out+0x3deaae)
2021-04-13T11:54:23.6029126Z     #13 0x55fa25e3a1bc  (/tmp/rustdoctestne7jWm/rust_out+0x3e21bc)
2021-04-13T11:54:23.6029814Z     #14 0x55fa25eb6e79  (/tmp/rustdoctestne7jWm/rust_out+0x45ee79)
2021-04-13T11:54:23.6030477Z     #15 0x55fa25f653e9  (/tmp/rustdoctestne7jWm/rust_out+0x50d3e9)
2021-04-13T11:54:23.6031152Z     #16 0x55fa25b77080  (/tmp/rustdoctestne7jWm/rust_out+0x11f080)
2021-04-13T11:54:23.6031727Z     #17 0x55fa25bb2c12  (/tmp/rustdoctestne7jWm/rust_out+0x15ac12)
2021-04-13T11:54:23.6032062Z 
2021-04-13T11:54:23.6032365Z Thread T1 (Worker 1) created by T0 here:
2021-04-13T11:54:23.6032859Z     #0 0x55fa25b3500c  (/tmp/rustdoctestne7jWm/rust_out+0xdd00c)
2021-04-13T11:54:23.6033421Z     #1 0x55fa26046249  (/tmp/rustdoctestne7jWm/rust_out+0x5ee249)
2021-04-13T11:54:23.6033984Z     #2 0x55fa25cdd135  (/tmp/rustdoctestne7jWm/rust_out+0x285135)
2021-04-13T11:54:23.6034547Z     #3 0x55fa25cded48  (/tmp/rustdoctestne7jWm/rust_out+0x286d48)
2021-04-13T11:54:23.6035119Z     #4 0x55fa25d19719  (/tmp/rustdoctestne7jWm/rust_out+0x2c1719)
2021-04-13T11:54:23.6035675Z     #5 0x55fa25c181ca  (/tmp/rustdoctestne7jWm/rust_out+0x1c01ca)
2021-04-13T11:54:23.6036247Z     #6 0x55fa25d02717  (/tmp/rustdoctestne7jWm/rust_out+0x2aa717)
2021-04-13T11:54:23.6036968Z     #7 0x55fa25ce4111  (/tmp/rustdoctestne7jWm/rust_out+0x28c111)
2021-04-13T11:54:23.6037565Z     #8 0x55fa25c94ed9  (/tmp/rustdoctestne7jWm/rust_out+0x23ced9)
2021-04-13T11:54:23.6038173Z     #9 0x55fa25c489e3  (/tmp/rustdoctestne7jWm/rust_out+0x1f09e3)
2021-04-13T11:54:23.6038766Z     #10 0x55fa25c9178b  (/tmp/rustdoctestne7jWm/rust_out+0x23978b)
2021-04-13T11:54:23.6039402Z     #11 0x55fa25d20cb5  (/tmp/rustdoctestne7jWm/rust_out+0x2c8cb5)
2021-04-13T11:54:23.6040004Z     #12 0x55fa25d22a60  (/tmp/rustdoctestne7jWm/rust_out+0x2caa60)
2021-04-13T11:54:23.6040613Z     #13 0x55fa25d20376  (/tmp/rustdoctestne7jWm/rust_out+0x2c8376)
2021-04-13T11:54:23.6041372Z     #14 0x55fa25c240a2  (/tmp/rustdoctestne7jWm/rust_out+0x1cc0a2)
2021-04-13T11:54:23.6042003Z     #15 0x55fa25c2623c  (/tmp/rustdoctestne7jWm/rust_out+0x1ce23c)
2021-04-13T11:54:23.6042619Z     #16 0x55fa25d22c10  (/tmp/rustdoctestne7jWm/rust_out+0x2cac10)
2021-04-13T11:54:23.6043254Z     #17 0x55fa25c08e77  (/tmp/rustdoctestne7jWm/rust_out+0x1b0e77)
2021-04-13T11:54:23.6043869Z     #18 0x55fa25d23001  (/tmp/rustdoctestne7jWm/rust_out+0x2cb001)
2021-04-13T11:54:23.6044498Z     #19 0x55fa25c08ccd  (/tmp/rustdoctestne7jWm/rust_out+0x1b0ccd)
2021-04-13T11:54:23.6045116Z     #20 0x55fa25c935f9  (/tmp/rustdoctestne7jWm/rust_out+0x23b5f9)
2021-04-13T11:54:23.6045741Z     #21 0x55fa25d1f16f  (/tmp/rustdoctestne7jWm/rust_out+0x2c716f)
2021-04-13T11:54:23.6046354Z     #22 0x55fa25c40f09  (/tmp/rustdoctestne7jWm/rust_out+0x1e8f09)
2021-04-13T11:54:23.6046976Z     #23 0x55fa25bb18ba  (/tmp/rustdoctestne7jWm/rust_out+0x1598ba)
2021-04-13T11:54:23.6047588Z     #24 0x55fa25b7e41d  (/tmp/rustdoctestne7jWm/rust_out+0x12641d)
2021-04-13T11:54:23.6048212Z     #25 0x55fa25b771e3  (/tmp/rustdoctestne7jWm/rust_out+0x11f1e3)
2021-04-13T11:54:23.6048833Z     #26 0x55fa25ff9c7d  (/tmp/rustdoctestne7jWm/rust_out+0x5a1c7d)
2021-04-13T11:54:23.6049478Z     #27 0x55fa25e36aae  (/tmp/rustdoctestne7jWm/rust_out+0x3deaae)
2021-04-13T11:54:23.6050115Z     #28 0x55fa25e3a1bc  (/tmp/rustdoctestne7jWm/rust_out+0x3e21bc)
2021-04-13T11:54:23.6050756Z     #29 0x55fa25eb6e79  (/tmp/rustdoctestne7jWm/rust_out+0x45ee79)
2021-04-13T11:54:23.6051374Z     #30 0x55fa25f653e9  (/tmp/rustdoctestne7jWm/rust_out+0x50d3e9)
2021-04-13T11:54:23.6051992Z     #31 0x55fa25b77080  (/tmp/rustdoctestne7jWm/rust_out+0x11f080)
2021-04-13T11:54:23.6052602Z     #32 0x55fa25bb2c12  (/tmp/rustdoctestne7jWm/rust_out+0x15ac12)
2021-04-13T11:54:23.6052967Z 
2021-04-13T11:54:23.6053770Z SUMMARY: AddressSanitizer: heap-use-after-free (/tmp/rustdoctestne7jWm/rust_out+0x342600) 
2021-04-13T11:54:23.6054493Z Shadow bytes around the buggy address:
2021-04-13T11:54:23.6054938Z   0x0c087fff83b0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6055417Z   0x0c087fff83c0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6055937Z   0x0c087fff83d0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6056452Z   0x0c087fff83e0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6056912Z   0x0c087fff83f0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6058518Z =>0x0c087fff8400: fa fa 00 00 00 00 00 fa fa fa fd fd fd fd[fd]fa
2021-04-13T11:54:23.6058949Z   0x0c087fff8410: fa fa 00 00 00 00 00 fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6059400Z   0x0c087fff8420: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6059852Z   0x0c087fff8430: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6060319Z   0x0c087fff8440: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6060774Z   0x0c087fff8450: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-04-13T11:54:23.6061348Z Shadow byte legend (one shadow byte represents 8 application bytes):
2021-04-13T11:54:23.6061837Z   Addressable:           00
2021-04-13T11:54:23.6062252Z   Partially addressable: 01 02 03 04 05 06 07 
2021-04-13T11:54:23.6062644Z   Heap left redzone:       fa
2021-04-13T11:54:23.6063119Z   Freed heap region:       fd
2021-04-13T11:54:23.6063503Z   Stack left redzone:      f1
2021-04-13T11:54:23.6063982Z   Stack mid redzone:       f2
2021-04-13T11:54:23.6064414Z   Stack right redzone:     f3
2021-04-13T11:54:23.6064795Z   Stack after return:      f5
2021-04-13T11:54:23.6065184Z   Stack use after scope:   f8
2021-04-13T11:54:23.6065613Z   Global redzone:          f9
2021-04-13T11:54:23.6065981Z   Global init order:       f6
2021-04-13T11:54:23.6066412Z   Poisoned by user:        f7
2021-04-13T11:54:23.6066801Z   Container overflow:      fc
2021-04-13T11:54:23.6067227Z   Array cookie:            ac
2021-04-13T11:54:23.6067599Z   Intra object redzone:    bb
2021-04-13T11:54:23.6068019Z   ASan internal:           fe
2021-04-13T11:54:23.6068408Z   Left alloca redzone:     ca
2021-04-13T11:54:23.6068844Z   Right alloca redzone:    cb
2021-04-13T11:54:23.6069210Z   Shadow gap:              cc
2021-04-13T11:54:23.6069559Z ==5349==ABORTING
2021-04-13T11:54:23.6069790Z 
2021-04-13T11:54:23.6069987Z 
2021-04-13T11:54:23.6070309Z 
2021-04-13T11:54:23.6070589Z failures:
2021-04-13T11:54:23.6071383Z     src/actor_ref/rpc.rs - actor_ref::rpc (line 86)
2021-04-13T11:54:23.6071648Z 
2021-04-13T11:54:23.6072202Z test result: FAILED. 47 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 30.32s
</code></pre>
</details>

Another log:

```log
failures:

---- src/actor_ref/rpc.rs - actor_ref::rpc (line 86) stdout ----
Test executable failed (exit code 1).

stdout:
Increased count to 10

stderr:
=================================================================
==5724==ERROR: AddressSanitizer: heap-use-after-free on address 0x604000002070 at pc 0x5612313b56b1 bp 0x7fee3a1fb540 sp 0x7fee3a1fb538
WRITE of size 1 at 0x604000002070 thread T2 (Sync actor 1000)
    #0 0x5612313b56b0  (/tmp/rustdoctestBzJQJc/rust_out+0x35a6b0)
    #1 0x56123118cf5e  (/tmp/rustdoctestBzJQJc/rust_out+0x131f5e)
    #2 0x5612311ad646  (/tmp/rustdoctestBzJQJc/rust_out+0x152646)
    #3 0x561231187c55  (/tmp/rustdoctestBzJQJc/rust_out+0x12cc55)
    #4 0x56123117553a  (/tmp/rustdoctestBzJQJc/rust_out+0x11a53a)
    #5 0x5612311a03c8  (/tmp/rustdoctestBzJQJc/rust_out+0x1453c8)
    #6 0x5612311b49b8  (/tmp/rustdoctestBzJQJc/rust_out+0x1599b8)
    #7 0x5612311b39de  (/tmp/rustdoctestBzJQJc/rust_out+0x1589de)
    #8 0x561231196ad4  (/tmp/rustdoctestBzJQJc/rust_out+0x13bad4)
    #9 0x56123119596b  (/tmp/rustdoctestBzJQJc/rust_out+0x13a96b)
    #10 0x56123117cafe  (/tmp/rustdoctestBzJQJc/rust_out+0x121afe)
    #11 0x56123117f926  (/tmp/rustdoctestBzJQJc/rust_out+0x124926)
    #12 0x5612311708e9  (/tmp/rustdoctestBzJQJc/rust_out+0x1158e9)
    #13 0x5612311801f2  (/tmp/rustdoctestBzJQJc/rust_out+0x1251f2)
    #14 0x5612311b8acc  (/tmp/rustdoctestBzJQJc/rust_out+0x15dacc)
    #15 0x56123117fe8a  (/tmp/rustdoctestBzJQJc/rust_out+0x124e8a)
    #16 0x56123117d3a9  (/tmp/rustdoctestBzJQJc/rust_out+0x1223a9)
    #17 0x56123117f2c8  (/tmp/rustdoctestBzJQJc/rust_out+0x1242c8)
    #18 0x561231183cb5  (/tmp/rustdoctestBzJQJc/rust_out+0x128cb5)
    #19 0x5612315ab529  (/tmp/rustdoctestBzJQJc/rust_out+0x550529)
    #20 0x5612315ab0a3  (/tmp/rustdoctestBzJQJc/rust_out+0x5500a3)
    #21 0x5612316376b8  (/tmp/rustdoctestBzJQJc/rust_out+0x5dc6b8)
    #22 0x7fee3db27608  (/lib/x86_64-linux-gnu/libpthread.so.0+0x9608)
    #23 0x7fee3d8f9292  (/lib/x86_64-linux-gnu/libc.so.6+0x122292)

0x604000002070 is located 32 bytes inside of 40-byte region [0x604000002050,0x604000002078)
freed by thread T1 (Worker 1) here:
    #0 0x5612311508f2  (/tmp/rustdoctestBzJQJc/rust_out+0xf58f2)
    #1 0x56123147dcac  (/tmp/rustdoctestBzJQJc/rust_out+0x422cac)
    #2 0x5612314ee41b  (/tmp/rustdoctestBzJQJc/rust_out+0x49341b)
    #3 0x5612311a968e  (/tmp/rustdoctestBzJQJc/rust_out+0x14e68e)
    #4 0x5612311a4665  (/tmp/rustdoctestBzJQJc/rust_out+0x149665)
[0m[0m[1m[31merror[0m[1m:[0m test failed, to rerun pass '--doc'
    #5 0x56123118a41b  (/tmp/rustdoctestBzJQJc/rust_out+0x12f41b)
    #6 0x5612311822a6  (/tmp/rustdoctestBzJQJc/rust_out+0x1272a6)
    #7 0x5612311af4b8  (/tmp/rustdoctestBzJQJc/rust_out+0x1544b8)
    #8 0x5612311b57c1  (/tmp/rustdoctestBzJQJc/rust_out+0x15a7c1)
    #9 0x5612311b2fc7  (/tmp/rustdoctestBzJQJc/rust_out+0x157fc7)
    #10 0x561231180505  (/tmp/rustdoctestBzJQJc/rust_out+0x125505)
    #11 0x56123121716a  (/tmp/rustdoctestBzJQJc/rust_out+0x1bc16a)
    #12 0x5612311fe204  (/tmp/rustdoctestBzJQJc/rust_out+0x1a3204)
    #13 0x5612311fd11c  (/tmp/rustdoctestBzJQJc/rust_out+0x1a211c)
    #14 0x56123125c826  (/tmp/rustdoctestBzJQJc/rust_out+0x201826)
make: *** [Makefile:28: test_sanitiser] Error 101
    #15 0x5612312707bf  (/tmp/rustdoctestBzJQJc/rust_out+0x2157bf)
    #16 0x56123135ceea  (/tmp/rustdoctestBzJQJc/rust_out+0x301eea)
    #17 0x561231351c0b  (/tmp/rustdoctestBzJQJc/rust_out+0x2f6c0b)
    #18 0x5612313624db  (/tmp/rustdoctestBzJQJc/rust_out+0x3074db)
    #19 0x561231363729  (/tmp/rustdoctestBzJQJc/rust_out+0x308729)
    #20 0x561231363bec  (/tmp/rustdoctestBzJQJc/rust_out+0x308bec)
    #21 0x5612313626bb  (/tmp/rustdoctestBzJQJc/rust_out+0x3076bb)
    #22 0x56123135161e  (/tmp/rustdoctestBzJQJc/rust_out+0x2f661e)
    #23 0x56123127339e  (/tmp/rustdoctestBzJQJc/rust_out+0x21839e)
    #24 0x5612315ab0a3  (/tmp/rustdoctestBzJQJc/rust_out+0x5500a3)
    #25 0x5612316376b8  (/tmp/rustdoctestBzJQJc/rust_out+0x5dc6b8)
    #26 0x7fee3db27608  (/lib/x86_64-linux-gnu/libpthread.so.0+0x9608)

previously allocated by thread T1 (Worker 1) here:
    #0 0x561231150a3d  (/tmp/rustdoctestBzJQJc/rust_out+0xf5a3d)
    #1 0x56123147d813  (/tmp/rustdoctestBzJQJc/rust_out+0x422813)
    #2 0x5612314ee36e  (/tmp/rustdoctestBzJQJc/rust_out+0x49336e)
    #3 0x5612311a391b  (/tmp/rustdoctestBzJQJc/rust_out+0x14891b)
    #4 0x5612311a3c9a  (/tmp/rustdoctestBzJQJc/rust_out+0x148c9a)
    #5 0x5612311a96ea  (/tmp/rustdoctestBzJQJc/rust_out+0x14e6ea)
    #6 0x561231175035  (/tmp/rustdoctestBzJQJc/rust_out+0x11a035)
    #7 0x56123119fd1e  (/tmp/rustdoctestBzJQJc/rust_out+0x144d1e)
    #8 0x56123119e1cc  (/tmp/rustdoctestBzJQJc/rust_out+0x1431cc)
    #9 0x5612311b2fc7  (/tmp/rustdoctestBzJQJc/rust_out+0x157fc7)
    #10 0x561231180505  (/tmp/rustdoctestBzJQJc/rust_out+0x125505)
    #11 0x56123121716a  (/tmp/rustdoctestBzJQJc/rust_out+0x1bc16a)
    #12 0x5612311fe204  (/tmp/rustdoctestBzJQJc/rust_out+0x1a3204)
    #13 0x5612311fd11c  (/tmp/rustdoctestBzJQJc/rust_out+0x1a211c)
    #14 0x56123125c826  (/tmp/rustdoctestBzJQJc/rust_out+0x201826)
    #15 0x5612312707bf  (/tmp/rustdoctestBzJQJc/rust_out+0x2157bf)
    #16 0x56123135ceea  (/tmp/rustdoctestBzJQJc/rust_out+0x301eea)
    #17 0x561231351c0b  (/tmp/rustdoctestBzJQJc/rust_out+0x2f6c0b)
    #18 0x5612313624db  (/tmp/rustdoctestBzJQJc/rust_out+0x3074db)
    #19 0x561231363729  (/tmp/rustdoctestBzJQJc/rust_out+0x308729)
    #20 0x561231363bec  (/tmp/rustdoctestBzJQJc/rust_out+0x308bec)
    #21 0x5612313626bb  (/tmp/rustdoctestBzJQJc/rust_out+0x3076bb)
    #22 0x56123135161e  (/tmp/rustdoctestBzJQJc/rust_out+0x2f661e)
    #23 0x56123127339e  (/tmp/rustdoctestBzJQJc/rust_out+0x21839e)
    #24 0x5612315ab0a3  (/tmp/rustdoctestBzJQJc/rust_out+0x5500a3)
    #25 0x5612316376b8  (/tmp/rustdoctestBzJQJc/rust_out+0x5dc6b8)
    #26 0x7fee3db27608  (/lib/x86_64-linux-gnu/libpthread.so.0+0x9608)

Thread T2 (Sync actor 1000) created by T0 here:
    #0 0x56123113b00c  (/tmp/rustdoctestBzJQJc/rust_out+0xe000c)
    #1 0x561231637049  (/tmp/rustdoctestBzJQJc/rust_out+0x5dc049)
    #2 0x56123117e064  (/tmp/rustdoctestBzJQJc/rust_out+0x123064)
    #3 0x56123117fb16  (/tmp/rustdoctestBzJQJc/rust_out+0x124b16)
    #4 0x561231194dc3  (/tmp/rustdoctestBzJQJc/rust_out+0x139dc3)
    #5 0x561231193d60  (/tmp/rustdoctestBzJQJc/rust_out+0x138d60)
    #6 0x561231194297  (/tmp/rustdoctestBzJQJc/rust_out+0x139297)
    #7 0x56123119a8c2  (/tmp/rustdoctestBzJQJc/rust_out+0x13f8c2)
    #8 0x5612311b7ae5  (/tmp/rustdoctestBzJQJc/rust_out+0x15cae5)
    #9 0x5612311841dd  (/tmp/rustdoctestBzJQJc/rust_out+0x1291dd)
    #10 0x56123117d1e3  (/tmp/rustdoctestBzJQJc/rust_out+0x1221e3)
    #11 0x56123158c5fd  (/tmp/rustdoctestBzJQJc/rust_out+0x5315fd)
    #12 0x561231479f8e  (/tmp/rustdoctestBzJQJc/rust_out+0x41ef8e)
    #13 0x56123147dfcc  (/tmp/rustdoctestBzJQJc/rust_out+0x422fcc)
    #14 0x5612314ecb39  (/tmp/rustdoctestBzJQJc/rust_out+0x491b39)
    #15 0x561231589461  (/tmp/rustdoctestBzJQJc/rust_out+0x52e461)
    #16 0x56123117d080  (/tmp/rustdoctestBzJQJc/rust_out+0x122080)
    #17 0x5612311b8b02  (/tmp/rustdoctestBzJQJc/rust_out+0x15db02)

Thread T1 (Worker 1) created by T0 here:
    #0 0x56123113b00c  (/tmp/rustdoctestBzJQJc/rust_out+0xe000c)
    #1 0x561231637049  (/tmp/rustdoctestBzJQJc/rust_out+0x5dc049)
    #2 0x561231350225  (/tmp/rustdoctestBzJQJc/rust_out+0x2f5225)
    #3 0x561231351e38  (/tmp/rustdoctestBzJQJc/rust_out+0x2f6e38)
    #4 0x5612312712e9  (/tmp/rustdoctestBzJQJc/rust_out+0x2162e9)
    #5 0x5612311ca94a  (/tmp/rustdoctestBzJQJc/rust_out+0x16f94a)
    #6 0x56123125aa07  (/tmp/rustdoctestBzJQJc/rust_out+0x1ffa07)
    #7 0x561231304caf  (/tmp/rustdoctestBzJQJc/rust_out+0x2a9caf)
    #8 0x5612312cf879  (/tmp/rustdoctestBzJQJc/rust_out+0x274879)
    #9 0x56123130fbc3  (/tmp/rustdoctestBzJQJc/rust_out+0x2b4bc3)
    #10 0x5612312cccdb  (/tmp/rustdoctestBzJQJc/rust_out+0x271cdb)
    #11 0x5612312e24b5  (/tmp/rustdoctestBzJQJc/rust_out+0x2874b5)
    #12 0x5612312e48b0  (/tmp/rustdoctestBzJQJc/rust_out+0x2898b0)
    #13 0x5612312e21c6  (/tmp/rustdoctestBzJQJc/rust_out+0x2871c6)
    #14 0x5612311f5ba9  (/tmp/rustdoctestBzJQJc/rust_out+0x19aba9)
    #15 0x5612311f7d5b  (/tmp/rustdoctestBzJQJc/rust_out+0x19cd5b)
    #16 0x5612312e4a51  (/tmp/rustdoctestBzJQJc/rust_out+0x289a51)
    #17 0x5612311bb038  (/tmp/rustdoctestBzJQJc/rust_out+0x160038)
    #18 0x5612312e4e2e  (/tmp/rustdoctestBzJQJc/rust_out+0x289e2e)
    #19 0x5612311bae9c  (/tmp/rustdoctestBzJQJc/rust_out+0x15fe9c)
    #20 0x5612312ceb50  (/tmp/rustdoctestBzJQJc/rust_out+0x273b50)
    #21 0x5612313335da  (/tmp/rustdoctestBzJQJc/rust_out+0x2d85da)
    #22 0x5612311f8054  (/tmp/rustdoctestBzJQJc/rust_out+0x19d054)
    #23 0x5612311b7730  (/tmp/rustdoctestBzJQJc/rust_out+0x15c730)
    #24 0x5612311841dd  (/tmp/rustdoctestBzJQJc/rust_out+0x1291dd)
    #25 0x56123117d1e3  (/tmp/rustdoctestBzJQJc/rust_out+0x1221e3)
    #26 0x56123158c5fd  (/tmp/rustdoctestBzJQJc/rust_out+0x5315fd)
    #27 0x561231479f8e  (/tmp/rustdoctestBzJQJc/rust_out+0x41ef8e)
    #28 0x56123147dfcc  (/tmp/rustdoctestBzJQJc/rust_out+0x422fcc)
    #29 0x5612314ecb39  (/tmp/rustdoctestBzJQJc/rust_out+0x491b39)
    #30 0x561231589461  (/tmp/rustdoctestBzJQJc/rust_out+0x52e461)
    #31 0x56123117d080  (/tmp/rustdoctestBzJQJc/rust_out+0x122080)
    #32 0x5612311b8b02  (/tmp/rustdoctestBzJQJc/rust_out+0x15db02)

SUMMARY: AddressSanitizer: heap-use-after-free (/tmp/rustdoctestBzJQJc/rust_out+0x35a6b0)
Shadow bytes around the buggy address:
  0x0c087fff83b0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c087fff83c0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c087fff83d0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c087fff83e0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c087fff83f0: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
=>0x0c087fff8400: fa fa 00 00 00 00 00 fa fa fa fd fd fd fd[fd]fa
  0x0c087fff8410: fa fa 00 00 00 00 00 fa fa fa fa fa fa fa fa fa
  0x0c087fff8420: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c087fff8430: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c087fff8440: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c087fff8450: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==5724==ABORTING



failures:
    src/actor_ref/rpc.rs - actor_ref::rpc (line 86)

test result: FAILED. 47 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 36.97s
```

But still can't find the cause.

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -4,6 +4,7 @@ env:
   RUST_BACKTRACE: full
 task:
   name: FreeBSD
+  timeout_in: 15m
   cargo_cache:
     folder: $HOME/.cargo/registry
   setup_script:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ test = ["getrandom"]
 
 [dependencies]
 crossbeam-channel = { version = "0.5.0", default-features = false, features = ["std"] }
-heph-inbox        = { version = "0.1.2", default-features = false }
+heph-inbox        = { version = "0.2.0", default-features = false }
 libc              = { version = "0.2.79", default-features = false }
 log               = { version = "0.4.8", default-features = false }
 mio               = { version = "0.7.5", default-features = false, features = ["os-poll", "tcp", "udp", "pipe"] }

--- a/src/test.rs
+++ b/src/test.rs
@@ -156,9 +156,8 @@ where
         Ok(())
     });
     waker
-        .block_on(receiver.recv())
+        .block_on(receiver.recv_once())
         .expect("failed to receive result from test runtime")
-        .0
 }
 
 /// Attempt to spawn a thread-local actor on the *test* runtime.

